packages/mds-policy-author/api.ts

```diff
@@ -108,49 +106,7 @@ const featureCollectionSchema = Joi.object()
   .unknown(true) // TODO
 
 function api(app: express.Express): express.Express {
-  /**
-   * Policy-specific middleware to extract provider_id into locals, do some logging, etc.
-   */
-  app.use(async (req: PolicyApiRequest, res: PolicyApiResponse, next: express.NextFunction) => {
-    res.header('Access-Control-Allow-Origin', '*')
-    try {
-      // TODO verify presence of agency_id
-      if (!(req.path.includes('/health') || req.path === '/' || req.path === '/schema/policy')) {
-        if (res.locals.claims) {
-          const { provider_id } = res.locals.claims
-
-          // TODO alter authorization code to look for an agency_id
-
-          if (provider_id) {
-            if (!isUUID(provider_id)) {
-              await log.warn(req.originalUrl, 'invalid provider_id is not a UUID', provider_id)
-              return res.status(400).send({
-                result: `invalid provider_id ${provider_id} is not a UUID`
-              })
-            }
-
-            if (!isProviderId(provider_id)) {
-              return res.status(400).send({
-                result: `invalid provider_id ${provider_id} is not a known provider`
-              })
-            }
-
-            if (![TEST1_PROVIDER_ID, TEST2_PROVIDER_ID, TEST4_PROVIDER_ID].includes(provider_id)) {
-              return res.status(401).send({ result: 'Unauthorized' })
-            }
-          }
-        } else {
-          return res.status(401).send('Unauthorized')
-        }
-      }
-    } catch (err) {
-      /* istanbul ignore next */
-      await log.error(req.originalUrl, 'request validation fail:', err.stack)
-    }
-    next()
-  })
-
-  app.get(pathsFor('/policies'), async (req, res) => {
+  app.get(pathsFor('/policies'), checkScope(check => check('policies:read')), async (req, res) => {
```

Comment:
What would the scope be for a provider to read the policies that are applicable to them? `policies:read:provider`?

Reply:
This isn't currently in the spec at all, but it's something I think needs to definitely be considered.

Reply:
Yes, that's the intent of `policies:read:provider` ... its enforcement would be a little different than just `checkScope` though.